### PR TITLE
chore: implement automated release workflow - part 2

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- implement automated release workflow - part 2

**Related issue(s)**

- https://github.com/kyma-project/control-plane/issues/3330

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -130,3 +130,161 @@ jobs:
...
+        id: pull-ref
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git checkout main
+          git stash
+          git pull
+          LATEST_COMMIT=$(git rev-parse HEAD)
+          echo "latest_commit=$LATEST_COMMIT" >> $GITHUB_OUTPUT
+          echo "Latest commit ref $LATEST_COMMIT"
```
Comment (k15r) [inline .github/workflows/create-release.yml:239]: is this something other than this?
```
git fetch origin
LATEST_COMMIT=$(git rev-parse origin/main)
```

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -130,3 +130,161 @@ jobs:
...
+    needs: [ create-draft, run-unit-tests, bump-kmc-chart ]
+    if: ${{ !inputs.dry-run }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ needs.bump-kmc-chart.outputs.latest_commit}}
```
Comment (k15r) [inline .github/workflows/create-release.yml:255]: still confused why we do not directly checkout `main` here

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -130,3 +130,161 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ needs.bump-sec-scanners-config.outputs.latest_commit}}
+      - run: |
+          git checkout main
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply .github/workflows/create-release.yml]: updated

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (mfaizanse) [reply .github/workflows/create-release.yml]: I have updated the ref test the behaviour.

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (mfaizanse) [reply .github/workflows/create-release.yml]: updated

**Review by marcobebway — APPROVED** [review]

**Pull request merged**